[CY-4890] Add support for `eslint-config-xo-typescript`

https://github.com/xojs/eslint-config-xo-typescript

Hi @twschiller we are facing a problem bumping the version of typescript by adding your configuration, based on the requirements `>=4.3` from https://github.com/xojs/eslint-config-xo-typescript/blob/main/package.json#L47

We currently have our typescript version as 4.2.4 that seems to run into some incompatibilities when it gets updated to 4.3.5. 

Is it possible for you to have your requirement as >=4.2 on your config? Also, feel free to contribute with a PR that solves the problems, like bellow when building the docker:

```node_modules/@types/jsforce/query.d.ts(70,13): error TS2368: Type parameter name cannot be 'never'.```

Otherwise we will want to tackle this eventually, just don't have an ETA for you.

--- a/package.json
+++ b/package.json
@@ -152,6 +152,7 @@
     "eslint-config-xo": "^0.38.0",
     "eslint-config-xo-react": "^0.25.0",
     "eslint-config-xo-space": "^0.29.0",
+    "eslint-config-xo-typescript": "^0.44.0",
     "eslint-import-resolver-alias": "^1.1.2",
     "eslint-import-resolver-babel-module": "^5.3.1",
     "eslint-import-resolver-node": "^0.3.6",

